src/journal/compress.c

```diff
@@ -541,6 +645,90 @@ int compress_stream_lz4(int fdf, int fdt, uint64_t max_bytes) {
 #endif
 }
 
+int compress_stream_zstd(int fdf, int fdt, uint64_t max_bytes) {
+#if HAVE_ZSTD
+        _cleanup_(ZSTD_freeCCtxp) ZSTD_CCtx *s = ZSTD_createCStream();
+        size_t out_size = ZSTD_CStreamOutSize();
+        uint8_t out[out_size];
+        char *src = NULL;
+        struct stat st;
+        ZSTD_inBuffer ZSTD_in = {src, 0, 0};
+        ZSTD_outBuffer ZSTD_out = {&out, sizeof(out), 0};
+        size_t ret;
+        uint64_t left = max_bytes;
+        int r;
+
+        assert(fdf >= 0);
+        assert(fdt >= 0);
+
+        if (!s)
+                return -ENOMEM;
+
+        ret = ZSTD_initCStream(s, ZSTD_COMPRESSION_LEVEL);
```

Comment:
This won't write the content size in the frame, so `decompress_blob_zstd()` can't decompress this data, since `size` will be `ZSTD_CONTENTSIZE_UNKNOWN`. Will data compressed with this function ever be decompressed with `decompress_blob_zstd()`?